### ISSUE TITLE
[Suggestion w. Sample Code] Encode [Flags] enums to integers so that bitmasks are preserved in-flight

### ISSUE THREAD

**Issue by zaneclaes** [ChilliCream/graphql-platform#6781]
### Product

Hot Chocolate

### Is your feature request related to a problem?

In a sense; there's some broken behavior around transport of special enums ([Flags] bitmasks).

Right now, AFAICT, if I have this enum...

```
[Flags]
public enum Test {
  Foo = 1 << 0,
  Bar = 1 << 1,
  Baz = 1 << 2,
}
```

I can run into problems with bitmasking and HTTP transport. To wit, `DefaultNamingConventions.GetEnumValueName(Test.Foo | Test.Bar)` returns an integer value (3), which is not allowed in terms of schema definitions. It doesn't explode the schema-generation because all enum values are well-defined, but when I try to use the values the response serializer fails:

```
      "message": "Test cannot serialize the given value.",
```

### The solution you'd like

I tried to tweak this with naming conventions at first, before I realized that did not affect the in-flight values. However, this is sample code showing what I was trying to do with turning flag-based enums into values that do not explode the API.

```
public class EnumFlagsNamingConventions : DefaultNamingConventions {
  public override string GetEnumValueName(object value) {
    var enumType = value.GetType();

    if (enumType.IsEnum && enumType.GetCustomAttribute<FlagsAttribute>() != null) {
      // Encode enums as numbers so that bitmask flags are preserved
      var str = ((int) value).ToString();
      string pre = "P"; // Enum values need to start with a string... "P" for Plus...
      if (str.StartsWith("-")) {
        // There should be no case where bitmask enums are negative, so this could probably be excluded
        pre = "M";
        str = str.Substring(1);
      }
      return pre + str;
    }
    return base.GetEnumValueName(value);
  }
}
```

It yields a schema like...
```
enum Foo {
  P1
  P2
  P4
}
```

... this is less readable in-flight, true, but it is now more predictable for a client to decode. To wit, a value of `P3` will be transmitted over-the-wire for `Test.Foo | Test.Bar`, which allows the client to merely assign the integer value `3` to its enum implementation. 

# Discussion

Assuming we're in CSharp land and sharing code, this will work well. Other languages may find it harder to convert the bitmask, but I think the code above is at least an improvement on the current scenario (i.e., falling back on poorly defined integer values over the wire). That said, just an idea ;)

An alternate approach might be to special-treat these enums at the HotChocolate level such that they are simply TYPED as if they were integers......

**Comment by michaelstaib** [comment]
Did you enable flags support?

**Comment by zaneclaes** [comment]
If this is documented, I'm not sure where to find it. I searched the documentation directly, Google, etc. and found nothing relevant before I started down this path. I see a few references to enum flags in code discussions on GitHub, but I didn't dive all the way down the rabbit hole. 

Is there somewhere I should be looking for features like these? Release notes maybe? I thought I had done a thorough search, but it seems like I still often miss things like this with HotChocolate...

<img width="599" alt="Screenshot 2023-12-15 at 10 14 13 AM" src="https://github.com/ChilliCream/graphql-platform/assets/2084821/07f5c511-f739-4d37-bf6a-1615f17c60a3">
